cpp/src/strings/count_matches.hpp

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2021-2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <cudf/column/column.hpp>
+
+#include <rmm/cuda_stream_view.hpp>
+
+namespace cudf {
+
+class column_device_view;
```

Comment:
I think we should try to use forward declarations in headers more in order to avoid over-including headers inside of headers.
Regardless, the `column_device_view` is defined in a `.cuh` file in there is no reason to include the `.cuh` in this `.hpp` file just to get the class name. I'm just trying to make it easier for the compiler since this particular `.cuh` has alot of code in it now.